tests/test_resource_manager.rs

```diff
@@ -0,0 +1,220 @@
+use futures::StreamExt;
+use k8s_openapi::api::core::v1::{ConfigMap, Namespace, Pod, Secret};
+use kube::api::{Api, DeleteParams, PostParams};
+use serde_json::json;
+
+#[derive(Clone, Debug)]
+pub enum TestResource {
+    Secret(String),
+    ConfigMap(String),
+    Pod(String),
+}
+
+#[derive(Clone, Debug)]
+pub enum TestResourceSpec {
+    Secret(String, String, String),    // single value per secret for now
+    ConfigMap(String, String, String), // single value per cm for now
+}
+
+impl TestResourceSpec {
+    pub fn secret(resource_name: &str, value_name: &str, value: &str) -> Self {
+        Self::Secret(
+            resource_name.to_owned(),
+            value_name.to_owned(),
+            value.to_owned(),
+        )
+    }
+
+    pub fn config_map(resource_name: &str, value_name: &str, value: &str) -> Self {
+        Self::ConfigMap(
+            resource_name.to_owned(),
+            value_name.to_owned(),
+            value.to_owned(),
+        )
+    }
+}
+
+pub struct TestResourceManager {
+    namespace: String,
+    client: kube::Client,
+    resources: Vec<TestResource>,
+}
+
+impl Drop for TestResourceManager {
+    fn drop(&mut self) {
+        let resources = self.resources.clone();
+        let namespace = self.namespace.clone();
+        let t = std::thread::spawn(move || {
+            let mut rt =
+                tokio::runtime::Runtime::new().expect("Failed to create Tokio runtime for cleanup");
+            rt.block_on(clean_up_resources(resources, namespace))
+        });
+
+        let thread_result = t.join();
+        let cleanup_result = thread_result.expect("Failed to clean up WASI test resources");
+        cleanup_result.unwrap()
+    }
+}
+
+impl TestResourceManager {
+    pub async fn initialise(namespace: &str, client: kube::Client) -> anyhow::Result<Self> {
+        let namespaces: Api<Namespace> = Api::all(client.clone());
+        namespaces
+            .create(
+                &PostParams::default(),
+                &serde_json::from_value(json!({
+                        "apiVersion": "v1",
+                        "kind": "Namespace",
+                        "metadata": {
+                            "name": namespace
+                        },
+                        "spec": {}
+                }))?,
+            )
+            .await?;
+
+        // k8s seems to need a bit of time for namespace permissions to flow
+        // through the system.  TODO: make this less worse
```

Comment:
This could probably be checked in the status to make sure `phase` == `Active` like we do for other waits. Not a big deal for this PR though